app/resources/hardware.py

```diff
@@ -0,0 +1,101 @@
+from typing import List, Tuple
+
+from scheme import UUID
+
+from app import wrapper, m
+from models.service import Service
+from models.workload import Workload
+from resources.game_content import check_compatible, calculate_power, scale_resources, generate_scale, dict2tuple, turn
+from schemes import requirement_build, device_not_found
+
+
+@m.user_endpoint(path=["hardware", "build"], requires=requirement_build)
+def build(data: dict, user: str):
+    """
+    This endpoint you can test your build if the parts a compatible and what is required for a device
+    """
+    comp, message = check_compatible(data)
+    if not comp:
+        return message
+
+    performance: tuple = calculate_power(data)
+
+    return_message: dict = {"success": True, "performance": performance}
+
+    return return_message
+
+
+@m.user_endpoint(path=["hardware", "resources"], requires={"device_uuid": UUID()})
+def hardware_resources(data: dict, user: str):
+    wl: Workload = wrapper.session.query(Workload).filter_by(uuid=data["device_uuid"])
+
+    if wl is None:
+        return device_not_found
+
+    return wl.display()
+
+
+@m.microservice_endpoint(path=["hardware", "register"])
+def hardware_register(data: dict, microservice: str):
+    # cpu_requirements, ram_req, gpu_req, disk_req, network_req
+
+    wl: Workload = wrapper.session.query(Workload).filter_by(uuid=data["device_uuid"])
```

Comment:
```suggestion
    wl: Workload = wrapper.session.query(Workload).filter_by(uuid=data["device_uuid"]).first()
```

Reply:
or we use wrapper.session.query(Workload).get(data["device_uuid"])
... it automatically queries for the primary key